### PR TITLE
server/mux: Make sure gracefulTimeout is honored properly.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
This was reproduced by removing the Read deadlne timeout.
Once this was removed the `m.currentReqs` would always
be active `1` because net.Conn is remembered by net/rpc
after hijacking the incoming `CONNECT` request.

Now if `m.currenReqs` is `1` then the server-mux.go code

```go
for {
       select {
       case <-time.After(m.gracefulTimeout):
	       return nil
       case <-ticker.C:
	       if atomic.LoadInt32(&m.currentReqs) <= 0 {
	               return nil
	       }
       }
```

Here `time.After()` never gets control since `m.currentReqs` is `1`.
Now if a user presses `CTRL-C` then the server process would never
die and just hangs.

## Motivation and Context
Found this issue while debugging #4476

## How Has This Been Tested?
Manually.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: cmd/server-mux.go [hunk 1]
```diff
@@ -511,16 +511,23 @@ func (m *ServerMux) Close() error {
 
 	// Starting graceful shutdown. Check if all requests are finished
 	// in regular interval or force the shutdown
-	ticker := time.NewTicker(serverShutdownPoll)
+	ticker := time.NewTicker(globalServerShutdownPoll)
 	defer ticker.Stop()
+
+	// Total elasped shutdown poll duration, after which we
+	// need to break out of this loop.
+	var totalElaspedShutdownPoll time.Duration
```
Comment (poornas) [inline cmd/server-mux.go]: typo - Elapsed

**Review by balamurugana — COMMENTED** [review]
The PR https://github.com/minio/minio/pull/4217 already fixes many bugs like this.

**Comment by harshavardhana** [comment]
> The PR #4217 already fixes many bugs like this.

@balamurugana i added a review, this PR issue is not addressed yet.

**Comment by harshavardhana** [comment]
Closing this PR as @balamurugana mentioned this is being fixed in #4217

**Pull request closed without merging**